lib/external.ts

```diff
@@ -0,0 +1,45 @@
+import * as Promise from 'bluebird';
+import { DockerProgress } from 'docker-progress';
+import * as Dockerode from 'dockerode';
+import * as _ from 'lodash';
+
+import { BuildTask } from './build-task';
+import { BuildProcessError } from './errors';
+import { LocalImage } from './local-image';
+
+const hasImageTag = (name: string): boolean => {
+	const tagRegex = /.+:[^/]+$/;
```

Comment:
You need to anchor the regex to the start of string like so: `/^.+:[^/]+$/` otherwise it will match anywhere (which given that the input is not arbitrary is slightly redundant but nice to enforce things where we can).